x/programs/rust/wasmlanche-sdk/src/state.rs

```diff
@@ -38,56 +38,40 @@ pub enum Error {
     Delete,
 }
 
-pub struct State<K>
-where
-    K: Into<Key> + Hash + PartialEq + Eq + Clone,
-{
-    cache: HashMap<K, Vec<u8>>,
+pub struct State<'a, K: Key> {
+    cache: &'a RefCell<HashMap<K, Vec<u8>>>,
```

Comment:
If programs were multi-threaded, I think it would be a lot harder to guarantee determinism. I don't think it's something we're going to have to worry about in the near future. 

Also, `RefCell` isn't `Sync`, you can still send it across threads
https://doc.rust-lang.org/stable/std/cell/struct.RefCell.html#impl-Send-for-RefCell%3CT%3E